utils/awsutils/session.go

```diff
@@ -96,19 +97,30 @@ func CreateSession(config *SessionConfig) (*session.Session, error) {
 	})
 }
 
-func NewSessionConfig(destinationConfig map[string]interface{}, timeout time.Duration, serviceName string) (*SessionConfig, error) {
-	if destinationConfig == nil {
-		return nil, errors.New("destinationConfig should not be nil")
+func NewSessionConfigForDestination(destination *backendconfig.DestinationT, timeout time.Duration, serviceName string) (*SessionConfig, error) {
+	if destination == nil {
+		return nil, errors.New("destination should not be nil")
 	}
 	sessionConfig := SessionConfig{}
-	if err := mapstructure.Decode(destinationConfig, &sessionConfig); err != nil {
+	if err := mapstructure.Decode(destination.Config, &sessionConfig); err != nil {
 		return nil, fmt.Errorf("unable to populate session config using destinationConfig: %w", err)
 	}
+	if sessionConfig.Region == "" {
```

Comment:
just need a confirmation that this is really required :)

Reply:
mostly not required as streaming dest handlers have region as mandatory param. S3 is where I think we are making region optional but in s3 we are not using this method. In one of the destination it was there so I pull it here but can be removed.

Reply:
just want to make sure that there is no possibility for any existing configuration to break due to this. Maybe using the backtesting tool would be enough here?

Reply:
Sure will do that